File: backend/app/assets/javascripts/data/match.ts
import {Player} from "./player"
import {Type} from "./type"
import {Group} from "./group"
import {MatchType} from "./matchType"

export interface Match {
	id?: number;
    startTime?: Date;
    isPlayed?: boolean;
    result?: any;
    isPlayable?: boolean;
    isInWaitingList?:boolean;
    table?:number[];
    team1?: Player[];
    team2?: Player[];
    matchType?: MatchType;
    type?: Type;
    group?: Group;
}

File: management-backend/src/main/kotlin/de/ttt/management/registration/domain/match/Match.kt
package de.ttt.management.registration.domain.match

import de.ttt.management.registration.domain.player.Player
import java.time.LocalDateTime

class Match(
    var id: Long? = null,
    var player1: Player? = null,
    var player2: Player? = null,
    var winner: Player? = null,
    var playGroup: PlayGroup? = null,
    var matchType: MatchType? = null,
    var result: String? = null,
    var resultRaw: String? = null,
    var sets1: Int? = null,
    var sets2: Int? = null,
    var balls1: Int? = null,
    var balls2: Int? = null,
    var isPlayed: Boolean = false,
    var isPlaying: Boolean = false,
    var nr: Int? = null,
    var roundNumber: Int? = null,
    var startTime: LocalDateTime? = null
)
